Model/SepaDirectDebitPaymentMethod.php

```diff
@@ -0,0 +1,228 @@
+            ],
+            'bic' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_bic',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_enable_bic'),
+            ],
+            'paymentAction' => [
+                'type'        => 'select',
+                'field'       => 'oxpayments__wdoxidee_transactionaction',
+                'options'     => TransactionModel::getTranslatedActions(),
+                'title'       => Helper::translate('wd_config_payment_action'),
+                'description' => Helper::translate('wd_config_payment_action_desc'),
+            ],
+            'creditorId' => [
+                'type'        => 'text',
+                'field'       => 'oxpayments__wdoxidee_creditorid',
+                'title'       => Helper::translate('wd_config_creditor_id'),
+                'description' => Helper::translate('wd_config_creditor_id_desc'),
+            ],
+            'sepaMandateCustom' => [
+                'type'        => 'textarea',
+                'field'       => 'oxpayments__wdoxidee_sepamandatecustom',
+                'title'       => Helper::translate('wd_sepa_mandate'),
+            ],
+        ];
+
+        return array_merge(parent::getConfigFields(), $aAdditionalFields);
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.0.1
+     */
+    public function getPublicFieldNames()
+    {
+        return array_merge(
+            parent::getPublicFieldNames(),
+            ['descriptor', 'additionalInfo', 'paymentAction', 'deleteCanceledOrder', 'deleteFailedOrder']
+        );
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.0.1
+     */
+    public function getCheckoutFields()
+    {
+        $aCheckoutFields = [
+            'accountHolder' => [
+                'type' => 'text',
+                'title' => Helper::translate('wd_account_holder_title'),
+                'required' => true,
+            ],
+            'iban' => [
+                'type' => 'text',
+                'title' => Helper::translate('wd_iban'),
+                'required' => true,
+            ],
+        ];
+
+        if ($this->_oPayment->oxpayments__wdoxidee_bic->value) {
+            $aCheckoutFields = array_merge($aCheckoutFields, [
+                'bic' => [
+                    'type' => 'text',
+                    'title' => Helper::translate('wd_bic'),
+                ],
+            ]);
+        }
+
+        return $aCheckoutFields;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @param string $sAction
+     *
+     * @return PaymentMethod
+     *
+     * @since 1.0.1
+     */
+    public function getPostProcessingPaymentMethod($sAction)
+    {
+        if ($sAction === TransactionModel::ACTION_CREDIT) {
```

Comment:
please can you move this check into a isRefund() method? so its clear that for a refund the payment method is sepa ct